packages/aws-lambda/src/ssm.js

```diff
@@ -0,0 +1,143 @@
+/*
+ * (c) Copyright IBM Corp. 2022
+ * (c) Copyright Instana Inc. and contributors 2022
+ */
+
+'use strict';
+
+const ENV_NAME = 'INSTANA_SSM_PARAM_NAME';
+let fetchedValue;
+let envValue;
+let errorFromAWS;
+let initTimeoutInMs = 0;
+
+module.exports.reset = () => {
+  fetchedValue = null;
+  envValue = null;
+  errorFromAWS = null;
+  initTimeoutInMs = 0;
+};
+
+module.exports.isUsed = () => !!envValue;
+
+module.exports.validate = () => {
+  // NOTE: Always reset the env value, we do not wanne cache, because it might change
+  envValue = null;
+
+  const _envValue = process.env[ENV_NAME];
+
+  if (!_envValue || !_envValue.length) {
+    return false;
+  }
+
+  envValue = _envValue;
+
+  return envValue;
+};
+
+module.exports.init = ({ logger }) => {
+  let AWS;
+
+  // CASE: Customer did not set INSTANA_SSM_PARAM_NAME, skip
+  if (!exports.isUsed()) {
+    return;
+  }
+
+  // CASE: We already fetched the SSM value, skip
+  if (fetchedValue) {
+    return;
+  }
+
+  initTimeoutInMs = Date.now();
+
+  try {
+    /**
+     * From https://docs.aws.amazon.com/lambda/latest/dg/lambda-nodejs.html:
+     *
+     * Your code runs in an environment that includes the AWS SDK for JavaScript,
+     * with credentials from an AWS Identity and Access Management (IAM) role that you manage.
+     */
+    // eslint-disable-next-line
+    AWS = require('aws-sdk');
+
+    // https://docs.aws.amazon.com/lambda/latest/dg/configuration-envvars.html
+    AWS.config.update({ region: process.env.AWS_REGION });
+
+    const ssm = new AWS.SSM();
+    const params = {
+      Name: envValue,
+      // See https://docs.aws.amazon.com/cli/latest/reference/ssm/get-parameter.html#options
+      // We always expect a string here. Never a decrypt
```

Comment:
I am not 100% sure about that. Since the use case is storing the Instana agent key securely, we might also expect users to store them as a secure key?

Reply:
Yeah you are right. Will add support for decrypt too :) 

From https://docs.aws.amazon.com/kms/latest/developerguide/services-parameter-store.html

> To decrypt the parameter value before returning it, set the WithDecryption parameter of GetParameter to true. When you use WithDecryption, Parameter Store calls the AWS KMS Decrypt operation on your behalf to decrypt the parameter value. As a result, the GetParameter request returns the parameter with a plaintext parameter value, as shown in the following example.

Reply:
Do we need both? Or only encrypted? If we offer both, we probably need a way for users to tell us which one it is, right?

Reply:
Added discussion to the ticket. Waiting for reply

Reply:
Discussed: we want to support both

We might want to introduce an extra env variable e.g. `INSTANA_SSM_DECRYPTION=Boolean`

Any other suggestions?